src/client.ts

```diff
@@ -230,17 +230,28 @@ class OptimizelyReactSDKClient implements ReactSDKClient {
 
     this.userPromise = new Promise(resolve => {
       this.userPromiseResolver = resolve;
-    }).then(() => {
-      this.isUserReady = true;
-      return { success: true };
+    }).then(isUserResolved => {
+      if (isUserResolved) {
+        this.isUserReady = true;
+        return { success: true };
+      }
+      return { success: false, message: 'Failed to fetch qualified segments.' };
     });
 
     if (this._client) {
       this._client.onReady().then(() => {
         this.isClientReady = true;
       });
 
-      this.dataReadyPromise = Promise.all([this.userPromise, this._client!.onReady()]).then(() => {
+      this.dataReadyPromise = Promise.all([this.userPromise, this._client!.onReady()]).then(res => {
+        if (!res[0].success) {
+          return {
+            success: false,
+            reason: 'QUALIFIED_SEGMENTS_NOT_READY',
```

Comment:
What about more generic "USER_NOT_READY" here for future extension? 
`message` below will have details "segments not ready".